src/macros.rs

```diff
@@ -1223,7 +1223,7 @@ pub(crate) use unlikely;
 #[allow(unused_macros)]
 macro_rules! static_cast_i32 {
     ($v:expr_2021) => {
-        ::std::mem::transmute::<u32, i32>($v)
+        u32::cast_signed($v.wrapping_neg())
```

Comment:
The wrapping_neg should definitely not be here. The macro name gives no indication that it also negates the number. This would be highly misleading and potentially lead to bugs down the road.

Also `cast_signed` requires a newer rust version which might be OK but needs careful consideration as it is a backwards compatibility breaking change.